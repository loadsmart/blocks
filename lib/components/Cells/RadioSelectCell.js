"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_native_1 = require("react-native");
var res_1 = require("../../res");
var Radios_1 = require("../Radios");
var RadioSelectCell = /** @class */ (function (_super) {
    __extends(RadioSelectCell, _super);
    function RadioSelectCell() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RadioSelectCell.prototype.render = function () {
        var _a = this.props, selected = _a.selected, title = _a.title;
        var extraTitleStyle = {
            fontFamily: selected ? res_1.Fonts.SharpSansBold : res_1.Fonts.SharpSansMedium,
        };
        return (react_1.default.createElement(react_native_1.View, { style: [styles.container, this.props.style] },
            react_1.default.createElement(Radios_1.Radio, { selected: selected === true, style: styles.radio }),
            react_1.default.createElement(react_native_1.Text, { style: [styles.title, extraTitleStyle] }, title)));
    };
    return RadioSelectCell;
}(react_1.Component));
exports.default = RadioSelectCell;
var styles = react_native_1.StyleSheet.create({
    container: {
        flexDirection: 'row',
        alignItems: 'center',
        height: 48,
        backgroundColor: res_1.Colors.DarkGrey,
        paddingHorizontal: 16,
    },
    radio: {
        marginRight: 20,
    },
    title: {
        color: res_1.Colors.White,
    },
});
