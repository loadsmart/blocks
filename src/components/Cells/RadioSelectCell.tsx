import React, { Component } from 'react'
import { StyleSheet, Text, View, ViewProps } from 'react-native'
import { Colors, Fonts } from '../../res'
import { Radio } from '../Radios'

interface Props extends ViewProps {
  title: string
  selected?: boolean
}

export default class RadioSelectCell extends Component<Props> {
  public render() {
    const { selected, title, testID } = this.props
    const extraTitleStyle = {
      fontFamily: selected ? Fonts.SharpSansBold : Fonts.SharpSansMedium,
    }
    return (
      <View style={[styles.container, this.props.style]} {...{ testID }}>
        <Radio selected={selected === true} style={styles.radio} />
        <Text style={[styles.title, extraTitleStyle]}>{title}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: 48,
    backgroundColor: Colors.DarkGrey,
    paddingHorizontal: 16,
  },
  radio: {
    marginRight: 20,
  },
  title: {
    color: Colors.White,
  },
})
