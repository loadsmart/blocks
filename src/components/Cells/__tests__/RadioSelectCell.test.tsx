import React from 'react'
import renderer from 'react-test-renderer'
import RadioSelectCell from '../RadioSelectCell'
import { Text } from 'react-native'

describe('RadioSelectCell', () => {
  it('has a title', () => {
    const testRenderer = renderer.create(<RadioSelectCell title="Walter White" />)
    const title = testRenderer.root.findByType(Text)
    expect(title.props.children).toBe('Walter White')
  })

  it('matches snapshot when selected', () => {
    const testRenderer = renderer.create(<RadioSelectCell title="Walter White" selected />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  it('matches snapshot when not selected', () => {
    const testRenderer = renderer.create(<RadioSelectCell title="Walter White" />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
