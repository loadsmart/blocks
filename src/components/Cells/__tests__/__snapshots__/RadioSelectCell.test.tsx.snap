// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSelectCell matches snapshot when not selected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#666666",
        "flexDirection": "row",
        "height": 48,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "marginRight": 20,
        },
        Object {
          "borderColor": "#484a4c",
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Object {
          "fontFamily": "SharpSans-Medium",
        },
      ]
    }
  >
    Walter White
  </Text>
</View>
`;

exports[`RadioSelectCell matches snapshot when selected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#666666",
        "flexDirection": "row",
        "height": 48,
        "paddingHorizontal": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "marginRight": 20,
        },
        Object {
          "borderColor": "#14d64d",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 6,
            "height": 12,
            "width": 12,
          },
          Object {
            "backgroundColor": "#14d64d",
          },
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Object {
          "fontFamily": "SharpSans-Bold",
        },
      ]
    }
  >
    Walter White
  </Text>
</View>
`;
